sockets/bandwidth/bandwidth-test.c

```diff
@@ -0,0 +1,115 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <netdb.h>
+#include <unistd.h>
+#include <inttypes.h>
+#include <string.h>
+
+/****************************************
+        Author: Tim Wood
+        Author: Phil Lopreiato
+        with a little help from
+        http://beej.us/guide/bgnet/
+****************************************/
+
+#define MAX_PAYLOAD_SIZE 1024
+
+int main(int argc, char ** argv)
+{
+        char* server_port = "1234";
+        char* server_ip = "127.0.0.1";
+        char *message = malloc(MAX_PAYLOAD_SIZE);
+        int sockfd, rc;
+        int i;
+        struct addrinfo hints, *server;
+        int o;
+        int sleep_time = 1;
+
+        /* Command line args:
+                -p port
+                -h host name or IP
+                -m message to send
+        */
+        while ((o = getopt (argc, argv, "p:h:m:")) != -1) {
+                switch(o){
+                case 'p':
+                        server_port = optarg;
+                        break;
+                case 'h':
+                        server_ip = optarg;
+                        break;
+                case 'm':
```

Comment:
Since you aren't letting the user specify a string, let's remove this command line option.

Instead, you should add a parameter that lets the user control how long this will run for--maybe the number of 1KB messages to be sent, or 0 for infinite.